### DISCUSSION TITLE
Timeout for requests

### DISCUSSION THREAD

**Discussion by DaspawnW** [linkerd/linkerd2#5844] [Q&A]
Dear Linkerd team,

I'm currently a bit confused about the "default" timeouts that are available in Linkerd.
In our org we have the following scenario:
Kubernetes Cluster (old cloud), Kubernetes Cluster (new Cloud); we linked the two cluster via Linkerd Multicluster setup (Works perfectly fine, great job :-) ). But we see there timeouts of 10 seconds that in our opinion are related to Linkerd default timeouts.

On the other side I found this discussion: https://github.com/linkerd/linkerd2/discussions/4394 and the corresponding ticket: https://github.com/linkerd/website/issues/721 that explain that there are no default timeouts.

It would be really great if you can elaborate a bit about the Linkerd timeouts for communication in one cluster and for timeouts for the communication via Multicluster linking.

Thank you in advance,
Best regards,
 Björn

**Comment by mateiidavid** [comment]
Hey Bjorn,

I looked into this a bit to see if I can help.

* `request timeouts`: afaik if you don't have any ServiceProfiles configured with explicit timeouts, you shouldn't see any request timeouts. There used to be a default in here but it has since been removed, 2.10 should definitely not have that in, for example.
* `connection timeouts`: couldn't find a lot of info here, historically it seems the conn timeout was 1s but since then we've made it so that you can [configure it](https://github.com/linkerd/linkerd2/pull/4759). 
* `protocol detection timeouts`: the proxy tries to automatically detect the protocol (multiplexed on the tcp conn)

```
Linkerd 2.10 will also improve how protocol detection works by making it “fail open”: if the protocol detection code sees no client-side bytes after 10 seconds, it will treat the connection as a TCP connection and continue, rather than failing as in 2.9. This means that worst-case behavior of not annotating a server-speaks-first port with opaque-ports (or skip-ports) is a 10-second connect-time latency, rather than a connect failure.
(ref): [Protocol Detection and Opaque Ports in Linkerd](https://linkerd.io/2021/02/23/protocol-detection-and-opaque-ports-in-linkerd/)
```

Hope this helps, lmk if you have any questions!

**Reply by DaspawnW** [reply]
Thank you in advance for this detailed information :-)
I really appreciate this detailed answer
